src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatelessMarshallingStrategy.cs

```diff
@@ -295,6 +295,93 @@ public IEnumerable<StatementSyntax> GenerateCleanupStatements(TypePositionInfo i
+        }
+
+        public IEnumerable<StatementSyntax> GenerateGuaranteedUnmarshalStatements(TypePositionInfo info, StubCodeContext context) => _innerMarshaller.GenerateGuaranteedUnmarshalStatements(info, context);
+        public IEnumerable<StatementSyntax> GenerateMarshalStatements(TypePositionInfo info, StubCodeContext context) => _innerMarshaller.GenerateMarshalStatements(info, context);
+        public IEnumerable<StatementSyntax> GenerateNotifyForSuccessfulInvokeStatements(TypePositionInfo info, StubCodeContext context) => _innerMarshaller.GenerateNotifyForSuccessfulInvokeStatements(info, context);
+        public IEnumerable<StatementSyntax> GeneratePinnedMarshalStatements(TypePositionInfo info, StubCodeContext context) => _innerMarshaller.GeneratePinnedMarshalStatements(info, context);
+        public IEnumerable<StatementSyntax> GeneratePinStatements(TypePositionInfo info, StubCodeContext context) => _innerMarshaller.GeneratePinStatements(info, context);
+        public IEnumerable<StatementSyntax> GenerateSetupStatements(TypePositionInfo info, StubCodeContext context)
+        {
+            foreach (StatementSyntax statement in _innerMarshaller.GenerateSetupStatements(info, context))
+            {
+                yield return statement;
+            }
+
+            // bool <freeUnmanaged> = false;
+            yield return LocalDeclarationStatement(
+                VariableDeclaration(
+                    PredefinedType(Token(SyntaxKind.BoolKeyword)),
+                    SingletonSeparatedList(
+                        VariableDeclarator(
+                            Identifier(context.GetAdditionalIdentifier(info, "freeUnmanaged")),
+                            null,
+                            EqualsValueClause(
+                                LiteralExpression(SyntaxKind.FalseLiteralExpression))))));
+
+            // <nativeType> <original> = <nativeIdentifier>;
+            yield return LocalDeclarationStatement(
+                VariableDeclaration(
+                    AsNativeType(info).Syntax,
+                    SingletonSeparatedList(
+                        VariableDeclarator(
+                            Identifier(context.GetAdditionalIdentifier(info, "original")),
+                            null,
+                            EqualsValueClause(
+                                IdentifierName(context.GetIdentifiers(info).native))))));
+        }
+
+        public IEnumerable<StatementSyntax> GenerateUnmarshalCaptureStatements(TypePositionInfo info, StubCodeContext context)
+        {
+            foreach (StatementSyntax statement in _innerMarshaller.GenerateUnmarshalCaptureStatements(info, context))
+            {
+                yield return statement;
+            }
+
+            // Now that we've captured the new value to pass to the caller, we need to make sure that we free the old one.
+
+            // <freeUnmanaged> = true;
```

Comment:
What scenarios would we use this StatelessByRefFreeMarshalling type but not call GenerateUnmarshalCaptureStatements? Do we/could we have a test for one of those scenarios? In the generated code I can only find code that sets `freeUnmanaged` to false, then immediately to true again.